radio/src/storage/yaml/yaml_datastructs_128x64.cpp

```diff
@@ -81,7 +81,9 @@ const struct YamlIdStr enum_Functions[] = {
   {  FUNC_BACKLIGHT, "BACKLIGHT"  },
   {  FUNC_SCREENSHOT, "SCREENSHOT"  },
   {  FUNC_RACING_MODE, "RACING_MODE"  },
+#if defined(AUDIO_MUTE_GPIO)
   {  FUNC_DISABLE_AUDIO_AMP, "DISABLE_AUDIO_AMP"  },
+#endif
```

Comment:
I don't believe this is a valid method of doing this - this file should never be manually edited.

Reply:
are you sure? 2.9 and earlier had radio specific files. 2.10 changed this recently and combined bw radios into this more generic file. Too generic now. Is this file generated by some script?

Reply:
oops, saw it. note to self: search before coding and posting. and get to know the build process better.

Reply:
Been there, done that, have many a merit badge and "what am I doing wrong here" DMs to prove it 🤣

Reply:
I made the same mistake then in 2.9. revert the 2.9 commit? how to do it right?

Reply:
2.9 looks fine - it's stock `generate-yaml.sh`. This looks to be something different you added because of the single combined 128x64 file. Not sure what the process will need to be now.

Reply:
No,  not stock. I added the func entry to the radio files having the capability.

Reply:
Ah, that explains why T8 needed updating ;) I ran `generate-yaml.sh` just before merging the PR and that one also has `AUDIO_MUTE_GPIO_PIN` defined - it didn't pick up any other changes needed.

Reply:
sorry, didn't know about the generator. what I don't get is the generator is automatic only invoked at changes of datastructs.h, another data something file and changes to the template and the .py itself. It never picks up changes eg like adding an SF. Guess it's manually triggered then.

Better to revert 2.9. Sorry again.

back to the drawing board. I'll try to have the SF generally defined but exclude selecting the SF for non-capable  radios at UI level. This is what cpn does.

Reply:
`generate-yaml.sh` is manually run, which you would normally do when making changes to `datastructs_private.h`. Where you're coming unstuck here is to due to changes in 2.10, where all the 128x64 radios are being lumped together, so `#if defined(AUDIO_MUTE_GPIO)` in `datastructs_private.h` (like in 2.9) doesn't seem to be being handled correctly, since it's valid for some 128bw, but not others.

Reply:
Right, due to lumping the lump I added the `#if defined(AUDIO_MUTE_GPIO)` to `yaml_datastructs_128x64.cpp` not knowing (my bad) this is a generated file. I reverted the change to `yaml_datastructs_128x64.cpp`, added the func entry (without conditions) to all `yaml_datastructs_xxx.cpp` hoping this is what the generator would do now. Adding the SF to `bool isLogicalSwitchFunctionAvailable(int function)` should exclude the selection based on AUDIO_MUTE_GPIO.

Can you please do me a favor and trigger a build with newly generated yaml_datastructs?

Reply:
Done. Generator just put them in a different order on X10, X12, but otherwise ok.

Reply:
Thanks